### PR TITLE
Added more hooks for flowers

### PR DESCRIPTION
**Pull request by soundlogic2236**
Flowers can now control their light level and their redstone behavior.

### CODE REVIEW

File: src/main/java/vazkii/botania/common/Botania.java [hunk 1]
```diff
@@ -45,11 +51,20 @@ public void preInit(FMLPreInitializationEvent event) {
 		thaumcraftLoaded = Loader.isModLoaded("Thaumcraft");
 		bcTriggersLoaded = ModAPIManager.INSTANCE.hasAPI("BuildCraftAPI|statements");
 		bloodMagicLoaded = Loader.isModLoaded("AWWayofTime"); // Psh, noob
+		
+		if(lightHelper==null)
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/Botania.java]: SPACES SOUNDLOGIC SPACES

File: src/main/java/vazkii/botania/common/Botania.java [hunk 2]
```diff
@@ -45,11 +51,20 @@ public void preInit(FMLPreInitializationEvent event) {
...
+			lightHelper = new LightHelperVanilla();
 
 		proxy.preInit(event);
 	}
 
 	@EventHandler
+	@Optional.Method(modid = "easycoloredlights")
+	public void preInitColoredLights(FMLPreInitializationEvent event) {
+		lightHelper = new LightHelperColored();
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/Botania.java]: Did you test this? It looks like something that wouldn't work.

File: src/main/java/vazkii/botania/common/block/BlockFloatingSpecialFlower.java [hunk 3]
```diff
@@ -50,6 +52,39 @@ public BlockFloatingSpecialFlower() {
...
+		return true;
+	}
+
+	@Override
+	public int getComparatorInputOverride(World world, int x, int y, int z, int side) {
+		return ((TileSpecialFlower) world.getTileEntity(x, y, z)).getComparatorInputOverride(side);
+	}
+
+	@Override
+    public int isProvidingWeakPower(IBlockAccess world, int x, int y, int z, int side) {
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/block/BlockFloatingSpecialFlower.java]: Fix your indentation. There's an eclipse cleanup profile provider.

File: src/main/java/vazkii/botania/common/integration/coloredlights/ILightHelper.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package vazkii.botania.common.integration.coloredlights;
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/integration/coloredlights/ILightHelper.java]: Missing copyleft header.

File: src/main/java/vazkii/botania/common/integration/coloredlights/LightHelper.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package vazkii.botania.common.integration.coloredlights;
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/integration/coloredlights/LightHelper.java]: Missing copyleft header.

File: src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperColored.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package vazkii.botania.common.integration.coloredlights;
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperColored.java]: Missing copyleft header.

File: src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperVanilla.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package vazkii.botania.common.integration.coloredlights;
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperVanilla.java]: Missing copyleft header.

File: src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperColored.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+/*
+ * This is an implementation of ILightHelper for when the colored lights mod is installed
+ */
+public class LightHelperColored implements ILightHelper{
+
+	@Override
+	public int makeRGBLightValue(float r, float g, float b,
+			int currentLightValue) {
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperColored.java]: No newline.

File: src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperVanilla.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+/*
+ * This is an implementation of ILightHelper for when the colored lights mod isn't installed
+ */
+public class LightHelperVanilla implements ILightHelper{
+
+	@Override
+	public int makeRGBLightValue(float r, float g, float b,
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperVanilla.java]: No newline.

File: src/main/java/vazkii/botania/common/Botania.java (hunk 1 above)
Comment (soundlogic2236) [inline src/main/java/vazkii/botania/common/Botania.java]: I missed ONE? Grr...

File: src/main/java/vazkii/botania/common/Botania.java (hunk 2 above)
Comment (soundlogic2236) [inline src/main/java/vazkii/botania/common/Botania.java]: This was how they said to do it in the minecraftforge channel, and I can confirm that said code is called if and only if said mod is installed

File: src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperColored.java (hunk 8 above)
Comment (soundlogic2236) [inline src/main/java/vazkii/botania/common/integration/coloredlights/LightHelperColored.java]: I'm not clear what you mean by this.

**Pull request closed without merging**